### PR TITLE
ci: add rocky 9.1 builds configuration for azure

### PR DESCRIPTION
**Pull request by supershal**
**What problem does this PR solve?**:
- Add ubuntu 18.04 and rocky 9.1 e2e tests for azure.

**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-NUMBER

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by supershal** [comment]
We also have azure tests for "rhel-7", "oracle-7", "sles-15" and "flatcar" in teamcity which are not migrated over GHA.
These configurations are not supported by https://docs.d2iq.com/dkp/2.4/supported-infrastructure-operating-systems

Not sure why those builds added in Teamcity. Should we migrate them to GHA too? 
or should we disable them in TC?
@faiq @dkoshkin

**Review by dkoshkin — APPROVED** [review]

File: .github/workflows/azure-e2e.yaml [hunk 1]
```diff
@@ -13,6 +13,8 @@ jobs:
       matrix:
         os:
         - "ubuntu 20.04"
+        - "ubuntu 18.04"
```
Comment (faiq) [inline .github/workflows/azure-e2e.yaml:16]:
```suggestion
```

**Review by faiq — CHANGES_REQUESTED** [review]
we dont support ubuntu 18 on azure

**Comment by faiq** [comment]
> We also have azure tests for "rhel-7", "oracle-7", "sles-15" and "flatcar" in teamcity which are not migrated over GHA.
These configurations are not supported by https://docs.d2iq.com/dkp/2.4/supported-infrastructure-operating-systems
> Not sure why those builds added in Teamcity. Should we migrate them to GHA too?
or should we disable them in TC?
@faiq @dkoshkin

we shouldn't add builds that we don't claim we support. adding them to the previous CI was a mistake

File: .github/workflows/azure-e2e.yaml (hunk 1 above)
Comment (supershal) [reply .github/workflows/azure-e2e.yaml:16]: We support Ubuntu 18.04 for AKS. https://docs.d2iq.com/dkp/2.3/supported-infrastructure-operating-systems#id-(2.3)SupportedInfrastructureOperatingSystems-AKS 
However I notice that konvoy CLI does not use "gallery image" flag for aks cluster/nodepool create. 
so just to confirm, aks users will always use upstream azure image?

**Comment by DreadPirateShawn** [comment]
> we shouldn't add builds that we don't claim we support

I'm inclined to agree.

File: .github/workflows/azure-e2e.yaml (hunk 1 above)
Comment (dkoshkin) [reply .github/workflows/azure-e2e.yaml:16]: Correct DKP users will use the default upstream images for AKS (and EKS).
I agree with @faiq we should only add configs that DKP supports.

Comment (supershal) [reply .github/workflows/azure-e2e.yaml:16]: Thank you.

**Review by faiq — APPROVED** [review]

**Pull request merged**